rend3/src/datatypes.rs

```diff
@@ -393,6 +413,72 @@ impl Mesh {
         }
     }
 
+    /// Calculate tangents for the given mesh, based on normals and texture coordinates
+    pub fn calculate_tangents(&mut self) {
+        Self::calculate_tangents_for_buffers(
+            &mut self.vertex_tangents,
+            &self.vertex_positions,
+            &self.vertex_normals,
+            &self.vertex_uvs,
+            &self.indices,
+        );
+    }
+
+    fn calculate_tangents_for_buffers(
```

Comment:
My only big concern with this is that there may be case where verts really should be split into multiple (as kvark described). That being said I doubt it will affect most meshes, and we can deal with that in a future PR, doesn't have to be now.